### ISSUE TITLE
Pressing ctrl-c while autocomplete is in progress throws an exception

### ISSUE THREAD

**Issue by mitar** [kislyuk/argcomplete#262]
I have complicated CLI and it takes a while for autocomplete to work its magic. This is why I noticed that if I press `ctrl-c` just after I pressed `tab` I get an exception from autocomplete because (I am guessing) patching was not complete:

```
...
  File "/usr/local/lib/python3.6/dist-packages/argcomplete/__init__.py", line 223, in __call__
    completions = self._get_completions(comp_words, cword_prefix, cword_prequote, last_wordbreak_pos)
  File "/usr/local/lib/python3.6/dist-packages/argcomplete/__init__.py", line 255, in _get_completions
    completions = self.collect_completions(active_parsers, parsed_args, cword_prefix, debug)
  File "/usr/local/lib/python3.6/dist-packages/argcomplete/__init__.py", line 459, in collect_completions
    completions += self._get_option_completions(active_parser, cword_prefix)
  File "/usr/local/lib/python3.6/dist-packages/argcomplete/__init__.py", line 363, in _get_option_completions
    if not self._action_allowed(action, parser):
  File "/usr/local/lib/python3.6/dist-packages/argcomplete/__init__.py", line 373, in _action_allowed
    for conflict_action in parser._action_conflicts.get(action, []):
```

I think the issue is that argcomplete eats all exceptions, so it eats also `KeyboardInterrupt` exception.

I think the solution might be a custom handler for `ctrl-c` which would just exit immediately.

Not sure why bash even passes through `ctrl-c` to the autocomplete program.

**Comment by evanunderscore** [comment]
Is this printed to the terminal? By default, I'd expect all output from stdout and stderr to be suppressed. This redirection happens at the shell level, so it shouldn't matter what the program does. Can you supply a cut down script that reproduces this?

**Comment by evanunderscore** [comment]
I believe this is a fish issue and will be fixed by #338.
